src/Component.php

```diff
@@ -69,6 +69,19 @@ public function addComponent(self $component, $key = null)
         }
     }
 
+    /**
+     * Set all Components.
+     *
+     * @param array $components The array of Component that will be set
+     * @param null  $key        The key of the Component
+     */
+    public function setComponents(array $components)
```

Comment:
Yes but it comes with performance costs. I'm not sure this is necessary, it will fail eventually if one element is not a component. Let's see what @markuspoerschke thinks.

Reply:
There is a lot of type safety missing right now. To me it looks fine.